File: pkg/models/jsonschema/saved_filter.go
package jsonschema

import (
	"github.com/stashapp/stash/pkg/fsutil"
	"github.com/stashapp/stash/pkg/models"
)

type SavedFilter struct {
	Mode         models.FilterMode      `db:"mode" json:"mode"`
	Name         string                 `db:"name" json:"name"`
	FindFilter   *models.FindFilterType `json:"find_filter"`
	ObjectFilter map[string]interface{} `json:"object_filter"`
	UIOptions    map[string]interface{} `json:"ui_options"`
}

func (s SavedFilter) Filename() string {
	ret := fsutil.SanitiseBasename(s.Name + "_" + s.Mode.String())
	return ret + ".json"
}

func LoadSavedFilterFile(filePath string) (*SavedFilter, error) {
	return loadFile[SavedFilter](filePath)
}

func SaveSavedFilterFile(filePath string, image *SavedFilter) error {
	return saveFile[SavedFilter](filePath, image)
}

File: stash-rewrite/src/Stash.Core/Entities/SavedFilter.cs
using Stash.Core.Enums;

namespace Stash.Core.Entities;

public class SavedFilter : BaseEntity
{
    public FilterMode Mode { get; set; }
    public string Name { get; set; } = string.Empty;
    public string? FindFilter { get; set; } // JSON
    public string? ObjectFilter { get; set; } // JSON
    public string? UIOptions { get; set; } // JSON
}
